packages/2d/src/components/Text.ts

```diff
@@ -82,18 +92,37 @@ export class Text extends Shape {
     text: string,
     rect: Rect,
   ) {
+    const textDirection = this.textDirection();
+    const lang = this.lang();
+    const languageDirection = Text.rtlLanguages.includes(this.lang())
+      ? 'rtl'
+      : 'ltr';
+
     const y = rect.y + rect.height / 2;
+    let x = rect.x;
+
+    // Adjustments for the odd cases where textDirection is set to RTL but language itself
+    // is a LTR or vice versa.
+
+    if (lang != null && textDirection == 'ltr' && languageDirection == 'rtl') {
```

Comment:
Problem here is that if we do not supply the language direction, then we will have to detect it ourselves which is a slight overhead IMHO for the common case.